src/clj/game/cards/assets.clj

```diff
@@ -1736,11 +1736,20 @@
                               (play-cost-bonus [:credit 2]))}}}
 
    "Sundew"
-   {:implementation "it's all broken just don't even try man"}
-    ; :events {:runner-spent-click {:once :per-turn
-    ;                               :msg (req (when (not this-server) "gain 2 [Credits]"))
-    ;                               :effect (req (when (not this-server)
-    ;                                              (gain-credits state :corp 2)))}}
+   ; If this a run event then handle in :begin-run as do not know the server
+   ; being run on in :runner-spent-click.
+   {:events {:runner-spent-click {:once :per-turn
+                                  :req (req (and (first-event? state side :runner-spent-click)
+                                                 (not= :run (get-in @state [:runner :register :click-type]))))
+                                  :msg "gain 2 [Credits]"
+                                  :effect (req (gain-credits state :corp 2))}
+
+             :begin-run {:once :per-turn
+                         :req (req (and (first-event? state side :runner-spent-click)
+                                        (not this-server)))
```

Comment:
This needs to check that the `:click-type` is `:run`, because only Run subtype events trigger Sundew:
* [If the Runner plays a run event to initiate a run on the server Sundew is installed in, the Corp does not gain 2 credits. [Official FAQ]](https://netrunnerdb.com/en/card/25092)

Reply:
I don't know why I need to check it. It seems like I should know that this is a run because this is in the `:begin-run` event. Can you explain further?

Reply:
If you use Deuces Wild to make a run on the Sundew server click 1, Sundew still fires because technically you're not spending your first click making a run on the Sundew server, you're playing an event. Only events that have the "Run" subtype count as "run events" for Sundew.

You're only setting the `:click-type` if the click-run ability is used or if an event with a "Run" subtype is played, which is necessary to know if it should or should not block Sundew. Otherwise, literally any kind of run (Deuces Wild for instance) in the first click will block Sundew.

Reply:
Thanks for the info. I was not aware of the difference between a "run event" and an "event" that makes a run.

Reply:
No worries, I had to ask folks on Stimslack about it to make sure.